### PR TITLE
Restore compatibility in GPT-2 LoRALinear bias init

### LINKED ISSUE #484: Random numbers drawn after instantiating GPT2 model differ between `adapter-transformers` v3.1.0 and `transformers` v4.23.1
## Environment info
- `adapter-transformers` version: 3.1.0
- Platform: Linux WSL2-x86_64-with-glibc2.31
- Python version: 3.9.2
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 1.12.0+cu102 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

## Information

Model I am using (Bert, XLNet ...): GPT2

Language I am using the model on (English, Chinese ...): English

Adapter setup I am using (if any): None

The problem arises when using:
* [x] my own modified scripts: see script below

The tasks I am working on:
* [x] this applies to any task where extra, randomly initialized, layers (e.g. linear layers) are added on top of an imported model.

## To reproduce

Code to reproduce the behavior:

```
# initial checks: python version, transformers version, adapter-transformers version
import sys
import pkg_resources

print("python version:", sys.version)

try:
    pkg_resources.get_distribution("transformers").version
except pkg_resources.DistributionNotFound as err:
    print("transformers module not found")
else:
    print(
        f"transformers=={pkg_resources.get_distribution('transformers').version}",
    )

try:
    pkg_resources.get_distribution("adapter-transformers").version
except pkg_resources.DistributionNotFound as err:
    print("adapter-transformers module not found")
else:
    print(
        f"adapter-transformers=={pkg_resources.get_distribution('adapter-transformers').version}",
    )

# import modules
from transformers import AutoTokenizer, AutoModel
import torch

torch.manual_seed(42)
# generate some random input with torch
print(torch.randint(0, 5, (1, 10)))
# adapter-transformers==3.1.0: tensor([[2, 2, 1, 4, 1, 0, 0, 4, 0, 3]])
# adapter-transformers==3.0.0: tensor([[2, 2, 1, 4, 1, 0, 0, 4, 0, 3]])
# transformers==4.23.1:        tensor([[2, 2, 1, 4, 1, 0, 0, 4, 0, 3]])

tokenizer = AutoTokenizer.from_pretrained("gpt2")
print(torch.randint(0, 5, (1, 10)))
# adapter-transformers==3.1.0: tensor([[3, 4, 0, 4, 1, 2, 0, 0, 2, 1]])
# adapter-transformers==3.0.0: tensor([[3, 4, 0, 4, 1, 2, 0, 0, 2, 1]])
# transformers==4.23.1:        tensor([[3, 4, 0, 4, 1, 2, 0, 0, 2, 1]])

model = AutoModel.from_pretrained("gpt2")
print(torch.randint(0, 5, (1, 10)))
# adapter-transformers==3.1.0: tensor([[0, 1, 1, 3, 4, 1, 2, 4, 2, 3]])
# adapter-transformers==3.0.0: tensor([[0, 4, 3, 3, 2, 2, 3, 3, 2, 1]])
# transformers==4.23.1:        tensor([[0, 4, 3, 3, 2, 2, 3, 3, 2, 1]])
```

## Expected behavior

As `adapter-transformers` is a drop-in replacement of the `transformers` library, I expect random numbers (and randomly initialized layers added) to be the same between the two libraries, for a given random seed.

My guess is that `adapter-transformers` v3.1.0 code is drawing extra random values compared to `adapter-transformers` v3.0.0 or `transformers` v4.23.1.

### PR DESCRIPTION
**Pull request by calpt**
Fixes #484. Explanation in https://github.com/adapter-hub/adapter-transformers/issues/484#issuecomment-1485542062.

### CODE REVIEW

File: src/transformers/adapters/lora.py [hunk 1]
```diff
@@ -163,14 +163,19 @@ def __init__(
         config: PretrainedConfig,
         attn_key: str = None,
         fan_in_fan_out: bool = False,  # Set this to True if the layer to replace stores weight like (fan_in, fan_out)
+        no_init_bias: bool = False,  # Use this to add a bias that is not initialized by PyTorch
```
Comment (lenglaender) [inline src/transformers/adapters/lora.py:166]: I would rather add these comments as a docstring.

File: src/transformers/adapters/lora.py [hunk 2]
```diff
@@ -259,14 +264,19 @@ def __init__(
         out_features: int,
         location_key: str,
         config: PretrainedConfig,
-        fan_in_fan_out: bool = False,
+        fan_in_fan_out: bool = False,  # Set this to True if the layer to replace stores weight like (fan_in, fan_out)
+        no_init_bias: bool = False,  # Use this to add a bias that is not initialized by PyTorch
```
Comment (lenglaender) [inline src/transformers/adapters/lora.py:268]: Same thing as above

**Review by lenglaender — APPROVED** [review]
Looks good
